### ISSUE TITLE
Replicate appears to block node.js exit

### ISSUE THREAD

**Issue by yaronyg** [pouchdb/pouchdb#5271]
### Environment (Node.js/browser/hybrid app/etc.)

Node.js

### Browser/platform (Chrome/FF/Safari/Edge/iOS/Android/etc.)

N/A

### Adapter (IndexedDB/WebSQL/LevelDB/etc.)

LevelDB

### Server (CouchDB/Cloudant/Couchbase/PouchDB Server/etc.)

express-pouchdb

### Please describe your issue. Test cases (e.g. JSBin) are very helpful!

Run the code below on the command line using node and notice that even though the server close event and the replicate cancel event both have fired the node.js process still doesn't exit. What's keeping node.js open?

In my experiments if I have the replicate fail immediately then Node.js doesn't hang. For example, go below and manually set the serverPort variable to 44444 or some other bogus value and the code will exit just fine. Also note that I set pool to false (and even manually stepped into request.js to make sure it was honored, it is) to make sure this isn't a connection pooling issue.

What is going on?

```javascript
var PouchDB = require('pouchdb');
var express = require('express');
var expressPouchdb = require('express-pouchdb');
var http = require('http');

var dbPath = "foo";

var DefaultPouch = PouchDB.defaults({
  prefix: dbPath
});

var options = {
  skip_setup: false,
  ajax : {
    pool: false,
  }
};

var namePrefix = 'bar' + Date.now();

var app = express();
app.use('/db', expressPouchdb(DefaultPouch, {mode: 'minimumForPouchDB'}));
var server = http.createServer(app);
var socketcount = 0;
server.on('connection', function (socket) {
  ++socketcount;
  console.log(socketcount + ' open sockets');
  socket.on('close', function () {
    --socketcount;
    console.log('closed a socket, now have ' + socketcount);
  });
});

function putDocs(remoteDB) {
  var promises = [];
  for(var i = 0; i < 100; ++i) {
    promises.push(remoteDB.put({_id: 'stuff' + i}));
  }
  return Promise.all(promises);
}

server.listen(0, function () {
  var serverPort = server.address().port;
  var pouchDBRemoteUrl = "http://127.0.0.1:" + serverPort + "/db/" + namePrefix
    + "1";
  var remoteDB = new PouchDB(pouchDBRemoteUrl, options);
  var cancel = null;
  putDocs(remoteDB)
    .then(function () {
      var count = 0;
      return new Promise(function (resolve, reject) {
        cancel = remoteDB.replicate.to(namePrefix + '2', { live: true })
          .on('paused', function (err) {
            console.log('Got paused with ' + err);
          })
          .on('active', function () {
            console.log('Replication resumed');
          })
          .on('denied', function (err) {
            console.log('We got denied on a PouchDB access, this really should ' +
              'not happen - ' + err);
          })
          .on('complete', function (info) {
            console.log('complete - ' + JSON.stringify(info));
          })
          .on('error', function (err) {
            console.log('Got error on replication - ' + err);
            reject();
          })
          .on('change', function (info) {
            console.log('change - ' + JSON.stringify(info));
            count += info.docs_written;
            if (count === 100)
            {
              resolve();
            }
          })
          .on('cancel', function (info) {
            console.log('cancel - ' + info);
          });
      });
    })
    .catch(function (err) {
      console.log('catch got ' + err);
    })
    .then(function () {
      cancel && cancel.cancel();
      server.close(function () {
        console.log('I am closed!');
      });
    })
});
```

**Comment by daleharvey** [comment]
I am gonna suspect this is an issue in pouchdb-server, it does a lot of things to maintain couchdb compat and I know it has listeners that are started by default, any listener is gonna keep node open

**Comment by yaronyg** [comment]
So your hint led me to try an experiment, I turned live to false and suddenly everything cleaned up. So this made me think there is a problem in the continuous replication code which led me [here](https://github.com/pouchdb/express-pouchdb/blob/ef495aa4b5a9840400210adffba67af62aceb7af/lib/routes/changes.js#L32) which is continuing to be called even though there are no sockets left talking to the server (at least according to the socket counter I built into the demo) and the server is closed! 

In fact, I debugged and the socket associated with the res is marked as destroyed = true! So how the heck is the write not throwing some kind of exception. In any case, I'll move this bug to the proper repo.
